pkg/logql/metrics.go

```diff
@@ -138,6 +142,95 @@ func RecordMetrics(ctx context.Context, log log.Logger, p Params, status string,
 	recordUsageStats(queryType, stats)
 }
 
+func RecordLabelQueryMetrics(
+	ctx context.Context,
+	start, end time.Time,
+	label, status string,
+	stats stats.Result,
+) {
+	var (
+		logger      = util_log.WithContext(ctx, util_log.Logger)
+		latencyType = latencyTypeFast
+		queryType   = QueryTypeLabels
+	)
+
+	// Tag throughput metric by latency type based on a threshold.
+	// Latency below the threshold is fast, above is slow.
+	if stats.Summary.ExecTime > slowQueryThresholdSecond {
+		latencyType = latencyTypeSlow
+	}
+
+	// we also log queries, useful for troubleshooting slow queries.
+	level.Info(logger).Log(
+		"latency", latencyType,
+		"query_type", queryType,
+		"length", end.Sub(start),
+		"duration", time.Duration(int64(stats.Summary.ExecTime*float64(time.Second))),
+		"status", status,
+		"label", label,
+		"throughput", strings.Replace(humanize.Bytes(uint64(stats.Summary.BytesProcessedPerSecond)), " ", "", 1),
+		"total_bytes", strings.Replace(humanize.Bytes(uint64(stats.Summary.TotalBytesProcessed)), " ", "", 1),
+	)
+
+	bytesPerSecond.WithLabelValues(status, queryType, "", latencyType).
+		Observe(float64(stats.Summary.BytesProcessedPerSecond))
+	execLatency.WithLabelValues(status, queryType, "").
+		Observe(stats.Summary.ExecTime)
+	chunkDownloadLatency.WithLabelValues(status, queryType, "").
+		Observe(stats.ChunksDownloadTime().Seconds())
+	duplicatesTotal.Add(float64(stats.TotalDuplicates()))
+	chunkDownloadedTotal.WithLabelValues(status, queryType, "").
+		Add(float64(stats.TotalChunksDownloaded()))
+	ingesterLineTotal.Add(float64(stats.Ingester.TotalLinesSent))
+
+	// TODO(kavi): Do we need recordUsageStats for labels query?
```

Comment:
these are new to me, we use expvar? :thinking:

Reply:
yes. we expvar for the usage stats.

Reply:
with what though? how do we collect those? what's the reason they aren't just an internal go struct (if we collect them just from within our own go processes) or prometheus metrics if we need to see the values across processes?

Reply:
I'm not familiar with [usageStats](https://github.com/grafana/loki/tree/main/pkg/usagestats) either :). 

Removing the TODO, as it's not the scope of this PR to add usage stats to series and labels API. We can add it later in separate PR.